plugin/dnstap/dnstapio/io.go

```diff
@@ -52,10 +58,18 @@ func (dio *dnstapIO) Connect(endpoint string, socket bool) error {
 	if err != nil {
 		return err
 	}
-	go dio.serve()
+	dio.connected = true
```

Comment:
I this to be used from multiple threads? If so, this is a data race.

Reply:
Handling is in dio.serve(). It starts in Connect() only once.
Assume that we have Connect() on start CoreDNS and Close() on shutdown CoreDNS.
So all stuf is in one goroutine and we have no data race.